server.py

```diff
@@ -214,13 +217,107 @@ def _parse_languages(languages: str) -> Dict[str, int]:
     return ret
 
 
+def _parse_experience(experience: str) -> List[int]:
+    """
+    Return a string containing integers separated by commas as a list of ints.
+    """
+
+    ret = list()
+    if not experience:
+        return ret
+    for experience_level in experience.split(","):
+        ret.append(int(experience_level))
+    return ret
+
+
+def _get_coach_json(coach_args: JSON) -> CoachJSON:
+    """
+    Return a CoachJSON object corresponding the args from the get coaches
+    request.
+    """
+
+    # Turn all interger arguments into ints.
+    for int_arg in ["birth_year"]:
+        if int_arg in coach_args:
+            coach_args[int_arg] = int(coach_args[int_arg])
+
+    # Turn all list of interger arguments into lists of ints
+    for experience_args in ["need", "rights", "housing"]:
+        if experience_args in coach_args:
+            coach_args[experience_args] = _parse_experience(
+                coach_args[experience_args]
+            )
+
+    # Special case languages as it's a dict.
+    if "languages" in coach_args:
+        coach_args["languages"] = _parse_languages(coach_args["languages"])
+    
+    return coach_args
+
+
+def _include_coach(coach: CoachJSON, coach_filter: CoachJSON) -> bool:
+    """
+    Check whether to include a coach from the db in a repsoonse by filtering on
+    if the coach matches the fields in coach_filter.
+    """
+
+    if not coach_filter:
+        return True
```

Comment:
Get rid of this? `coach_filter` should not be `None`, and I'm assuming that's what it's intended to catch?

Reply:
`coach_filter` is None when called with the default arg, as in `get_coach_matches()`.

Reply:
Yeah, but I suggested changing the call to `if coach_filter and _coach_matches_filter(coach, coach_filter)`, because it doesn't make much sense to me for this function to accept a 'None' coach filter (which always returns True).

Reply:
The suggested change would result in nothing being returned if `coach_filter == None`. I'll change to the following:
```python
(coach_filter is None) or _coach_matches_filter(coach, coach_filter)
````

Reply:
Sorry yes, I meant `or` not `and` :)
Slight preference for `coach_filter or _coach_matches_filter(coach, coach_filter)` as a more pythonic way to write it, especially since `coach_filter=dict()` should be treated the same way as `coach_filter=None`, but meh.

Reply:
Changed